torch/_inductor/scheduler.py

```diff
@@ -1596,17 +1596,33 @@ def __init__(self, nodes: List[ir.Operation]) -> None:
             self.nodes = config._pre_fusion_custom_pass(self.nodes)
         self.nodes = self.fuse_nodes(self.nodes)
 
-        if self.post_grad_graph_id == 0:
-            # reorder forward graph
-            self.nodes = reorder.reorder_all_gather(
-                self.nodes, all_gather_before_last_wait=True
+        front_node = None
+        if config.simplefsdp.enable_bucket:
+            # get the first compute node w/o AG in backward graph
+            front_node = reorder.get_front_node(self.nodes)
+            # bucket all gather
+            self.nodes = transformer_block_bucket.bucket_all_gather_by_block(
+                self, self.nodes
             )
-        elif self.post_grad_graph_id == 1:
-            # reorder backward graph
-            self.nodes = reorder.reorder_all_gather(
-                self.nodes, all_gather_before_last_wait=False
-            )
-            self.nodes = reorder.reorder_reduce_scatter(self.nodes)
+
+            if self.post_grad_graph_id == 1:
```

Comment:
Let's figure out what's the best way to tell if a graph is forward/backward. We can ask Will on this. Maybe add a TODO item here if you can't figure out it very quickly.

Reply:
added